state-transition/core/state_processor_exits.go

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2025, Berachain Foundation. All rights reserved.
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package core
+
+import (
+	"github.com/berachain/beacon-kit/primitives/constants"
+	"github.com/berachain/beacon-kit/primitives/math"
+	"github.com/berachain/beacon-kit/primitives/version"
+	statedb "github.com/berachain/beacon-kit/state-transition/core/state"
+)
+
+// InitiateValidatorExit initiates the exit of the validator with index `idx`.
+func (sp *StateProcessor) InitiateValidatorExit(st *statedb.StateDB, idx math.ValidatorIndex) error {
+	validator, err := st.ValidatorByIndex(idx)
+	if err != nil {
+		return err
+	}
+	// By this point, the state should be correctly updated with the fork.
+	fork, err := st.GetFork()
+	if err != nil {
+		return err
+	}
```

Comment:
this is not necessarily true. Specifically this when this is called by < processValidatorSetCap < processEpoch < ProcessSlots. ProcessFork is always called _after_ ProcessSlots

Reply:
would rephrase comment to something like
```suggestion
	// We will use the fork version from the state to determine how to exit the validator.
	fork, err := st.GetFork()
	if err != nil {
		return err
	}
```

Reply:
noting: In the case the timestamp is between epochs (most likely), i.e. where first part of epoch is Deneb1, second part of epoch is Electra, we will process the validator exits at the end of this epoch as per the Electra rules.

And, if the timestamp of the Electra fork happens to be crossed at the first slot of a new epoch, we will process the exits of the previous epoch according to Deneb1 (this is bc ProcessSlots is called but ProcessFork hasn't been called yet). 

Correct me if I'm wrong on understanding this.

Reply:
> this is not necessarily true. Specifically this when this is called by < processValidatorSetCap < processEpoch < ProcessSlots. ProcessFork is always called after ProcessSlots

Very good point. Fixed. You're right in that it has implications around the fork boundary and worth a thought experiment. Will think on it and see if I can write a test for the described scenario

Reply:
I don’t think it is an issue though as there shouldn’t be any non determinism when replaying blocks

Reply:
Also it’s a ~1/192 chance of the latter case happening so i wouldn’t worry too much